### PR TITLE
Correction of _get_page function

### PR DESCRIPTION
**Pull request by silvavn**
The _get_page function was recursive. When too many errors happened it reaches stack overflow. I have changed it to be iterative.

Added fake user agent to randomise requests.

Overall code improvements.

Started PEP8 Compliance.

### CODE REVIEW

File: scholarly/scholarly.py [hunk 1]
```diff
@@ -108,54 +108,57 @@ def use_proxy(http='socks5://127.0.0.1:9050', https='socks5://127.0.0.1:9050'):
...
+                _HEADERS['User-Agent'] = UserAgent().random
+                resp = session.get(pagerequest,
+                                   headers=_HEADERS,
+                                   cookies=_COOKIES,
+                                   timeout=_TIMEOUT)
+                if resp.status_code == 200:
+                    if _has_captcha(resp.text):
+                        logger.info("Got a CAPTCHA. Retrying...")
+                    else:
+                        break
```
Comment (ipeirotis) [inline scholarly/scholarly.py:149]: This is risky. We get out of the loop with an incorrect response, but we are not going to refresh the Tor ID as we need an exception for the refresh to be triggered

Comment (ipeirotis) [reply scholarly/scholarly.py:149]: Perhaps raising an exception would be better.

Comment (silvavn) [reply scholarly/scholarly.py:151]: I have added an exception raise for consistency.
Also added *.log to gitignore.

Comment (ipeirotis) [reply scholarly/scholarly.py:151]: Need to raise an Exception also at 
```
logger.info(f"""Got a response code {resp.status_code}. Retrying...""")
```

**Pull request merged**